sdk/proto/provider.proto

```diff
@@ -59,6 +59,9 @@ service ResourceProvider {
     // Delete tears down an existing resource with the given ID.  If it fails, the resource is assumed to still exist.
     rpc Delete(DeleteRequest) returns (google.protobuf.Empty) {}
 
+    // Construct creates a new instance of the provided component resource and returns its state.
+    rpc Construct(ConstructRequest) returns (ConstructResponse) {}
```

Comment:
When `GetSchema` was added, users were seeing warnings when dynamic providers were used, because `GetSchema` was missing. Can we go ahead and add `Construct` as "unimplemented" to the dynamic providers as part of this change? See https://github.com/pulumi/pulumi/pull/4083

Reply:
Ah, will do 👍